### PR TITLE
Add spectate mode

### LINKED ISSUE #156: [FEATURE] Spectator Mode
## 🚀 Feature

Add a `View` button on the game list to take you the game as a spectator

### PR DESCRIPTION
**Pull request by BertovDev**
Closes #156 

## Motivation

Add spectate mode

## Summary of changes

Added new button.

## How has this been tested?

Start a game,and get kill.

## Checklist
- [x] I have tested the changes locally.
- [ ] I have tested the whole game after applying the changes, not only the affected areas.
- [ ] I self-reviewed the changes on GitHub, line by line.
- [ ] I have tested the changes in another devices.
  - [ ] Tested in iOS.
  - [ ] Tested in Android.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

**Review by AngieDutra — COMMENTED** [review]
Here is the visual reference @jazimp 

https://github.com/lambdaclass/curse_of_mirra/assets/82987608/68d696eb-5d24-4037-b501-6a0b03151481

**Review by jazimp — DISMISSED** [review]
I see these points to review in this implementation:

- The "defeated" text shouldn't overlap with the FPS and PING info.

- It's odd that the position of the "continue" button changes.

- Check that the typographic variable used in the buttons is consistent, (Source Sans 3 - Black).

**Comment by BertovDev** [comment]
Uploading Screen Recording 2024-05-09 at 6.54.46 PM.mov…

**Comment by jazimp** [comment]
visual approval ✅

**Comment by BertovDev** [comment]
https://github.com/lambdaclass/curse_of_mirra/assets/62818001/d4e2d83e-3d55-42c4-8b32-b40a80a7b554

References:

File: client/Assets/Resources/Fonts/SourceSans3-Bold SDF.asset
Comment (AngieDutra) [inline client/Assets/Resources/Fonts/SourceSans3-Bold SDF.asset:1]: This should not be pushed

**Review by AngieDutra — CHANGES_REQUESTED** [review]
Other than that is working as expected. It would be nice to add some animation otr transition opacity for level it up but for now it works and we can move on 👍🏼

**Review by AngieDutra — APPROVED** [review]
Tested on Android build and is working as expected

**Review by tvillegas98 — APPROVED** [review]
The spectate mode seems to work well!

**Pull request merged**